From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Aaron Kling <webgeek1234@gmail.com>
Date: Wed, 16 Apr 2025 20:56:20 -0500
Subject: [PATCH] Support prebuilt_kernel_headers

When a prebuilt kernel header archive is available, replace device, qti,
and generated kernel header deps with the prebuilt headers.

Change-Id: I6350aa6a731e008fa8e81e2455120c11d1040f6b

diff --git a/cc/cc.go b/cc/cc.go
index 4e0929ec3d50f9fcd312b5cbfeebd76c4307a440..bf9cc42710145d315ea837f4386a4b121179cee4 100644
--- a/cc/cc.go
+++ b/cc/cc.go
@@ -2943,8 +2943,11 @@ func (c *Module) DepsMutator(actx android.BottomUpMutatorContext) {
 			for _, entry := range list {
 				// Replace device_kernel_headers with generated_kernel_headers
 				// for inline kernel building
-				if entry == "device_kernel_headers" || entry == "qti_kernel_headers" {
-					if (ctx.Config().Getenv("INLINE_KERNEL_BUILDING") == "true") {
+				if entry == "device_kernel_headers" || entry == "qti_kernel_headers" || entry == "generated_kernel_headers" {
+					if (len(ctx.Config().Getenv("TARGET_PREBUILT_KERNEL_HEADERS")) > 0) {
+						newHeaderLibs = append(newHeaderLibs, "prebuilt_kernel_headers")
+						continue
+					} else if (ctx.Config().Getenv("INLINE_KERNEL_BUILDING") == "true" && entry != "generated_kernel_headers") {
 						newHeaderLibs = append(newHeaderLibs, "generated_kernel_headers")
 						continue
 					}
